torch/_dynamo/variables/optimizer.py

```diff
@@ -145,17 +160,26 @@ def wrap_tensor(self, tx, tensor_value):
         # if we have not seen a tensor before, stash and use a
         # global weak ref source, since it must be an optimizer tensor
         # that we have missed
+
         if tensor_value in self.tensor_to_source:
-            return VariableBuilder(tx, self.tensor_to_source[tensor_value])(
-                tensor_value
-            )
+            # mark these tensors as static for cudagraphs
```

Comment:
Should grads be marked as static ?

Reply:
Yeah u right

Reply:
just to be clear - i meant that the grads shouldn't be static, right ?

Reply:
They currently are not, they should be though.

My plan for this is to update zero_grad() to by default set static grads to zero rather than None, unless the user explicitly says I want grads to be none. I ran some experiments, and if the grads are set to zero they are in fact static across multiple iterations.

Reply:
If they are set to None, we'll get endless re-compilations here. Isn't that an issue ?

Reply:
Yeah I'll update default args of zero_grad to be `set_to_none=None` (vs true today). If set_to_none is None, we will set static grads to zero, and if it is True I will print a warning if there any static grads. Are you suggesting that I should just never set them to None? I'm following the principle of least surprise here

Reply:
discussed offline - so the POR is to have the user mark grads as static as an opt-in, not have dynamo do it automatically. If the upstream code is cudagraphed, the grads will already be static and work as intended. If the upstream code is eager, the user can mark the grads as static and zero_grad will not set them to `None`